src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/ProviderRelationship/ProviderRelationshipsService.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Client;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.Exceptions;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore.Projections.EditVacancyInfo;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using SFA.DAS.Http;
+using SFA.DAS.Http.TokenGenerators;
+
+namespace Esfa.Recruit.Vacancies.Client.Infrastructure.Services.ProviderRelationship
+{
+    public class ProviderRelationshipsService : IProviderRelationshipsService
+    {
+        private readonly ProviderRelationshipApiConfiguration _configuration;
+        private readonly ILogger<ProviderRelationshipApiConfiguration> _logger;
+        private readonly IJobsVacancyClient _jobsVacancyClient;
+
+        public ProviderRelationshipsService(IOptions<ProviderRelationshipApiConfiguration> configuration,
+            ILogger<ProviderRelationshipApiConfiguration> logger,
+            IJobsVacancyClient jobsVacancyClient)
+        {
+            _configuration = configuration.Value;
+            _logger = logger;
+            _jobsVacancyClient = jobsVacancyClient;
+        }
+
+        public async Task<IEnumerable<EmployerInfo>> GetLegalEntitiesForProviderAsync(long ukprn)
+        {
+            IEnumerable<EmployerInfo> employerInfos = new List<EmployerInfo>();
+            var httpClient = CreateHttpClient(_configuration);
+            var queryData = new { Ukprn = ukprn, Operation = "Recruitment" };            
+            var uri = new Uri(AddQueryString("accountproviderlegalentities", queryData), UriKind.RelativeOrAbsolute);
+
+            try
+            {
+                var response = await httpClient.GetAsync(uri);
+                if (response.IsSuccessStatusCode)
+                {
+                    var content  = await response.Content.ReadAsStringAsync();
+                    var relations = Newtonsoft.Json.JsonConvert.DeserializeObject<ProviderPermissions>(content);
+                    employerInfos =  MapEmployerInfo(relations);
+                    await AppendAddressToLegalEntitiesAsync(employerInfos);
+                }
+
+                _logger.LogError("An invalid response received when trying to get provider relationships");
```

Comment:
Will this LogError not get executed when it succeeds as well?

Reply:
Inverted if condition.